mslib/mscolab/server.py

```diff
@@ -53,9 +54,66 @@
 from mslib.utils import conditional_decorator
 from mslib.index import create_app
 from mslib.mscolab.forms import ResetRequestForm, ResetPasswordForm
+from mslib.mscolab import migrations
+
+
+def _handle_db_upgrade():
+    from mslib.mscolab.models import db
+
+    create_files()
+    inspector = sqlalchemy.inspect(db.engine)
+    existing_tables = inspector.get_table_names()
+    if ("alembic_version" not in existing_tables and len(existing_tables) > 0) or (
+        "alembic_version" in existing_tables
+        and len(existing_tables) > 1
+        and db.session.execute(sqlalchemy.text("SELECT * FROM alembic_version")).first() is None
+    ):
+        sys.exit(
+            """Your database contains no alembic_version revision identifier, but it has a schema. This suggests \
+that you have a pre-existing database but haven't followed the database migration instructions. To prevent damage to \
+your database MSColab will abort. Please follow the documentation for a manual database migration from MSColab v8/v9."""
+        )
```

Comment:
It would make sense for this error message to directly link to the migration documentation, but since our docs aren't versioned this seems impossible to do currently.